bit_manipulation/quine_mccluskey.cpp

```diff
@@ -0,0 +1,461 @@
+/**
+ * @file
+ * @author [jang-jaehyuk](https://github.com/jang-jaehyuk)
+ * @brief Implementation of [minimization of
+ * Boolean functions](https://en.wikipedia.org/wiki/Quine-McCluskey_algorithm)
+ * @details
+ * The Quine-McCluskey algorithm is an algorithm that minimizes logical expressions.
+ * The minimum form of a logical function can be obtained deterministically.
+ * f_{A,B,C,D}=BC'D'+AB'+AC
+ * The above expression minimizes the following expression and is logically equivalent.
+ * f_{A,B,C,D}=A'BC'D'+AB'C'D'+AB'C'D+AB'CD'+AB'CD+ABC'D'+ABCD'+ABCD
+ * 
+ * ### Algorithm
+ * 1. Find all the Prime Implicants of a given function.
+ * 2. Find the Essential Prime Implicant from the candidate term table using the candidate terms.
+ */
+
+#include <iostream>  /// for IO operations
+#include <vector>    /// for std::vector
+#include <algorithm> /// for sorting implicant
+#include <string>    /// for string operations
+#include <cassert>   /// for assert
+
+using std::cout;
+using std::vector;
+using std::cin;
+using std::endl;
+using std::min;
+using std::string;
+
+/**
+ * @brief This structure is for organizing the bits.
+ */
+typedef struct BitNode {
+	vector<int> v; /// combined numbers or selected number
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6)!** 😃